src/Github/Api/V3/CreateMilestoneThroughApiCall.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Github\Api\V3;
+
+use Laminas\AutomaticReleases\Git\Value\SemVerVersion;
+use Laminas\AutomaticReleases\Github\Value\RepositoryName;
+use Laminas\Diactoros\Uri;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\UriInterface;
+use Psr\Log\LoggerInterface;
+use Webmozart\Assert\Assert;
+
+use function Safe\json_decode;
+use function Safe\json_encode;
+use function sprintf;
+
+final class CreateMilestoneThroughApiCall implements CreateMilestone
+{
+    private const API_ROOT = 'https://api.github.com/';
+
+    private RequestFactoryInterface $messageFactory;
+
+    private ClientInterface $client;
+
+    /** @psalm-var non-empty-string */
+    private string $apiToken;
+
+    private LoggerInterface $logger;
+
+    /** @psalm-param non-empty-string $apiToken */
+    public function __construct(
+        RequestFactoryInterface $messageFactory,
+        ClientInterface $client,
+        string $apiToken,
+        LoggerInterface $logger
+    ) {
+        $this->messageFactory = $messageFactory;
+        $this->client         = $client;
+        $this->apiToken       = $apiToken;
+        $this->logger         = $logger;
+    }
+
+    public function __invoke(
+        RepositoryName $repository,
+        SemVerVersion $version
+    ): UriInterface {
+        $this->logger->info(sprintf(
+            'Creating milestone "%s" for "%s/%s"',
+            $version->fullReleaseName(),
+            $repository->owner(),
+            $repository->name()
+        ));
+
+        $request = $this->messageFactory
+            ->createRequest(
+                'POST',
+                self::API_ROOT . 'repos/' . $repository->owner() . '/' . $repository->name() . '/milestones'
+            )
+            ->withAddedHeader('Content-Type', 'application/json')
+            ->withAddedHeader('User-Agent', 'Ocramius\'s minimal API V3 client')
+            ->withAddedHeader('Authorization', 'token ' . $this->apiToken);
```

Comment:
Unrelated to this patch, but perhaps we should have a specialized PSR-17 request factory that adds these headers to the request generated by the factory? Thoughts, @Ocramius ?

Reply:
@weierophinney I don't think this is the responsibility of the factory, but rather of a Github API client: the fact that a token is added, and that a user-agent is added, is specific to Github's requirements, and we can decorate the request there if needed.

I would keep it as-is unless we keep adding more and more Github API interactions.